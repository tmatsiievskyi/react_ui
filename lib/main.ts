/* eslint-disable padding-line-between-statements */
export { Button } from './components/Button';
export { Input } from './components/Input';
export * from './components/Card';
