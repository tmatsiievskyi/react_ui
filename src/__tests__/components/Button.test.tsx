import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';

import { Button } from '../../../lib/main';

describe('Button', () => {
  it('should render Button', () => {
    render(<Button />);
    screen.debug();

    const el = screen.getByRole('button');

    expect(el).toBeInTheDocument();
  });

  it('renders correcty with buttonType prop', () => {
    const { container: containerPrimary } = render(
      <Button buttonType='primary' />,
    );

    const { container: containerSecondary } = render(
      <Button buttonType='secondary' />,
    );

    const { container: containerOutline } = render(
      <Button buttonType='outline' />,
    );

    render(
      <Button asChild buttonType='link'>
        <a href='http://link.c'>button link</a>
      </Button>,
    );

    expect(containerPrimary).toMatchSnapshot();
    expect(containerSecondary).toMatchSnapshot();
    expect(containerOutline).toMatchSnapshot();
    expect(screen.getByRole('link', { name: 'button link' })).toHaveAttribute(
      'href',
      'http://link.c',
    );

    expect(containerPrimary.firstChild).toHaveClass(
      `select-none bg-primary text-primary-foreground hover:bg-primary/90 transition-all 
          disabled:pointer-events-none disabled:opacity-70 disabled:shadow-none`,
    );
  });

  it('renders correctly with leftIcon prop', () => {
    const { container } = render(
      <Button label='With Left Icon' leftIcon={<span>ğŸ‘ˆ</span>} />,
    );

    expect(container).toHaveTextContent('ğŸ‘ˆ');
  });
});
