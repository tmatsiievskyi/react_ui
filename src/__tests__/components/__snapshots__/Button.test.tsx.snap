// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > renders correcty with buttonType prop 1`] = `
<div>
  <button
    class="flex flex-row gap-x-4 disabled:cursor-not-allowed items-center justify-center select-none bg-primary text-primary-foreground hover:bg-primary/90 transition-all disabled:pointer-events-none disabled:opacity-70 disabled:shadow-none text-base py-2 px-4 rounded-md"
  />
</div>
`;

exports[`Button > renders correcty with buttonType prop 2`] = `
<div>
  <button
    class="flex flex-row gap-x-4 disabled:cursor-not-allowed items-center justify-center select-none bg-secondary text-secondary-foreground shadow-sm hover:bg-secondary/80 hover:shadow-md transition-all text-base py-2 px-4 rounded-md"
  />
</div>
`;

exports[`Button > renders correcty with buttonType prop 3`] = `
<div>
  <button
    class="flex flex-row gap-x-4 disabled:cursor-not-allowed items-center justify-center select-none bg-background text-primary/90 hover:bg-accent/70 border border-primary/50 hover:text-accent-foreground text-base py-2 px-4 rounded-md"
  />
</div>
`;
